src/tarantool.client/LogicalConnectionManager.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using ProGaudi.Tarantool.Client.Model;
+using ProGaudi.Tarantool.Client.Model.Requests;
+using ProGaudi.Tarantool.Client.Model.Responses;
+using ProGaudi.Tarantool.Client.Utils;
+
+namespace ProGaudi.Tarantool.Client
+{
+    public class LogicalConnectionManager : ILogicalConnection
+    {
+        private readonly ClientOptions _clientOptions;
+
+        private readonly RequestIdCounter _requestIdCounter = new RequestIdCounter();
+
+        private LogicalConnection _droppableLogicalConnection;
+
+        private readonly ManualResetEvent _connected = new ManualResetEvent(false);
+
+        private readonly AutoResetEvent _reconnectAvailable = new AutoResetEvent(true);
+
+        private Timer _timer;
+
+        private int _disposing;
+
+        private const int _connectionTimeout = 1000;
+
+        private const int _pingTimerInterval = 100;
+
+        private int _pingCheckInterval = 1000;
+
+        private DateTimeOffset _nextPingTime = DateTimeOffset.MinValue;
+
+        public LogicalConnectionManager(ClientOptions options)
+        {
+            _clientOptions = options;
+        }
+
+        public void Dispose()
+        {
+            if (Interlocked.Exchange(ref _disposing, 1) > 0)
+            {
+                return;
+            }
+
+            Interlocked.Exchange(ref _droppableLogicalConnection, null)?.Dispose();
+            Interlocked.Exchange(ref _timer, null)?.Dispose();
+        }
+
+        public async Task Connect()
+        {
+            _clientOptions.LogWriter?.WriteLine($"{nameof(LogicalConnectionManager)}: Connecting...");
```

Comment:
using ReadWriteLock, not Slim - the only wait point is Connect and it is long, no reason for spin wait